core/app/models/spree/promotion/actions/create_line_items.rb

```diff
@@ -44,9 +44,13 @@ def perform(options = {})
             current_quantity = order.quantity_of(item.variant)
             next unless current_quantity < item.quantity && item_available?(item)
 
-            line_item = Spree::Dependencies.cart_add_item_service.constantize.call(order: order,
-                                                                                   variant: item.variant,
-                                                                                   quantity: item.quantity - current_quantity).value
+            line_item = Spree::Dependencies.cart_add_item_service.constantize.call(
+              order: order,
+              variant: item.variant,
+              quantity: item.quantity - current_quantity,
+              options: { validate_only_quantity: true }
```

Comment:
I added the possibility here to add a free item to the cart when it's in draft state